nrf52-hal-common/src/gpio.rs

```diff
@@ -51,21 +51,293 @@ pub enum Level {
 }
 
 
+// ===============================================================
+// Implement Generic Pins for this port, which allows you to use
+// other peripherals without having to be completely rust-generic
+// across all of the possible pins
+// ===============================================================
+/// Generic $PX pin
+pub struct Pin<MODE> {
+    pub pin: u8,
+    #[cfg(feature = "52840")]
+    pub port: bool,
```

Comment:
I'd probably prefer a `u8` to a bool, though probably not much difference in practice.

Reply:
Well, I wanted an u8 too. BUT: The svd2rust files are generated this way. they use a bool too, guess because it saw it's only 1 bit wide. So I left this as a bool so you can directly use the .port field when you want to set this when selecting pins etc.
Also, I highly doubt there will ever be a third port. The nRF52 is a low power small form factor chip. It already has a lot of pins for this usecase I think. And there is still room for 16 more pins on the second port.